# MINODES
MINODES: MARCH-INSIDE for Network Nodes

MINODES: MARCH-INSIDE for Networks Nodes is a software developed by C.R. Muteanu (algorithm and software design, software programing, and author of papers, https://github.com/muntisa/), H. González-Díaz (algorithm and software design, and author of papers, https://github.com/glezdiazh), and A. Duardo-Sánches (consulting on intellectual property issues and co-author of papers for social and legal networks analysis, https://github.com/aliuskaduardo) 

Algorithm origin:

MINDOES is based on MARCH-INSIDE: Markov Chain Invariants for Networks Simulation and Design MARCH-INSIDE (MI), a well-known method introduced by Prof. Humbert G Díaz (Gonzaléz-Díaz et al.) as early as 2002 for the calculation of Markov Invariants (Moments, Shanon entropies, Mean Markov values) of molecular graphs and complext netxorks using a Markov chain stchastic approach. MINODES extend MI algorithms ideas in order to calculate MI node centralities of complex networks. Some of these MI node centralities are Markov kth-node degrees, Markov kth-node clossenes, etc.; which are kth higher order analogues of classic node centralities now re-calculated and extended to higher order analogues using a MI approach. MINODES is able to read multiple files of complex networks (protein interaction networks, metabolic networks, social networks, etc.) in .mat, .net, and other formats and return MI node centralities of all nodes in the network. In case you want to develop new collaborations, applications, etc. related to MI algorith please do not hesitate to contact us at Linkedin: Prof. Humbert g. Díaz https://www.linkedin.com/in/humbertgdiaz/ and/or Prof. C.R. Munteanu https://www.linkedin.com/in/muntisa/.

Related Algorithms:

MARCH-INSIDE (MI) algorithm is the original algorithm https://github.com/glezdiazh/MARCH-INSIDE on which MINODES is based. Other algorithms based also on MI and then related to MINDOES somehow are: Sequence to Stars Networks (S2SNET) by C.R. Munteanu and González-Díaz H. https://github.com/muntisa/S2SNet); R-Markov Topological Indices (RMARKOVTI) by C.R. Munteanu https://github.com/muntisa/RMarkovTI, S2SNET Phyton (PyS2SNET) by C.R. Munteanu https://github.com/muntisa/pyS2SNet, etc.

MINODES Main authors contributions:

C.R. munteanu (algorithm and software design, software programing, AI/ML applications, main author of papers, https://github.com/muntisa/).

H. Gónzalez-Díaz (algorithm and software design, AI/ML applications, main author of papers, https://github.com/glezdiazh),

A. Duardo-Sánches (assistance with intellectual proprty issues and co-author of papers for social and legal networks analysis, https://github.com/aliuskaduardo)

Applications: 

MINODES, similarly to MI parameters, can be used to study the Markov Chain stochastic behaviour of graph or network-like systems, quantify the structure of complex biomo-lecular systems, and/or as input of Artificial Intelligence / Machine Learning (AI/ML) algorithms in order to seek predictive models. MI parameters have been to predict properties of small-sized drugs, proteins sequences, proteins 3D structures, RNA secondary structures, metabolic networks, criminal causality networks, biological networks, social networks, etc.


See references:

1: Duardo-Sánchez A, Munteanu CR, Riera-Fernández P, López-Díaz A, Pazos A,
González-Díaz H. Modeling complex metabolic reactions, ecological systems, and
financial and legal networks with MIANN models based on Markov-Wiener node
descriptors. J Chem Inf Model. 2014 Jan 27;54(1):16-29. doi: 10.1021/ci400280n.
Epub 2013 Dec 23. PMID: 24320872. 

2: González-Díaz H. Network topological indices, drug metabolism, and
distribution. Curr Drug Metab. 2010 May;11(4):283-4. doi:
10.2174/138920010791514162. PMID: 20545616.

3: Riera-Fernández P, Munteanu CR, Dorado J, Martin-Romalde R, Duardo-Sanchez A,
González-Diaz H. From chemical graphs in computer-aided drug design to general
Markov-Galvez indices of drug-target, proteome, drug-parasitic disease,
technological, and social-legal networks. Curr Comput Aided Drug Des. 2011
Dec;7(4):315-37. doi: 10.2174/157340911798260340. PMID: 22050683.

4: Riera-Fernández P, Martín-Romalde R, Prado-Prado FJ, Escobar M, Munteanu CR,
Concu R, Duardo-Sanchez A, González-Díaz H. From QSAR models of drugs to complex
networks: state-of-art review and introduction of new Markov-spectral moments
indices. Curr Top Med Chem. 2012;12(8):927-60. doi: 10.2174/156802612800166819.
PMID: 22352918.

5: Riera-Fernández P, Munteanu CR, Escobar M, Prado-Prado F, Martín-Romalde R,
Pereira D, Villalba K, Duardo-Sánchez A, González-Díaz H. New Markov-Shannon
Entropy models to assess connectivity quality in complex networks: from
molecular to cellular pathway, Parasite-Host, Neural, Industry, and Legal-Social
networks. J Theor Biol. 2012 Jan 21;293:174-88. doi: 10.1016/j.jtbi.2011.10.016.
Epub 2011 Oct 25. PMID: 22037044.

6: González-Díaz H, Riera-Fernández P. New Markov-autocorrelation indices for
re-evaluation of links in chemical and biological complex networks used in
metabolomics, parasitology, neurosciences, and epidemiology. J Chem Inf Model.
2012 Dec 21;52(12):3331-40. doi: 10.1021/ci300321f. Epub 2012 Nov 26. PMID:
23121444.

7: González-Díaz H, Arrasate S, Sotomayor N, Lete E, Munteanu CR, Pazos A,
Besada-Porto L, Ruso JM. MIANN models in medicinal, physical and organic
chemistry. Curr Top Med Chem. 2013;13(5):619-41. doi:
10.2174/1568026611313050006. PMID: 23548024.

8: González-Díaz H, Riera-Fernández P, Pazos A, Munteanu CR. The Rücker-Markov
invariants of complex Bio-Systems: applications in Parasitology and
Neuroinformatics. Biosystems. 2013 Mar;111(3):199-207. doi:
10.1016/j.biosystems.2013.02.006. Epub 2013 Feb 27. PMID: 23454544.

9: Vergara-Galicia J, Prado-Prado FJ, Gonzalez-Diaz H. Galvez-Markov network
transferability indices: review of classic theory and new model for
perturbations in metabolic reactions. Curr Drug Metab. 2014;15(5):557-64. doi:
10.2174/1389200215666140605125827. PMID: 24909421.

10: Gonzalez-Diaz H, Arrasate S, Juan AG, Sotomayor N, Lete E, Speck-Planche A,
Ruso JM, Luan F, Cordeiro MN. Matrix trace operators: from spectral moments of
molecular graphs and complex networks to perturbations in synthetic reactions,
micelle nanoparticles, and drug ADME processes. Curr Drug Metab.
2014;15(4):470-88. doi: 10.2174/1389200215666140908101604. PMID: 25204825.

11: González-Díaz H, González-Díaz Y, Santana L, Ubeira FM, Uriarte E.
Proteomics, networks and connectivity indices. Proteomics. 2008 Feb;8(4):750-78.
doi: 10.1002/pmic.200700638. PMID: 18297652.

12: González-Díaz H, Herrera-Ibatá DM, Duardo-Sánchez A, Munteanu CR, Orbegozo-
Medina RA, Pazos A. ANN multiscale model of anti-HIV drugs activity vs AIDS
prevalence in the US at county level based on information indices of molecular
graphs and social networks. J Chem Inf Model. 2014 Mar 24;54(3):744-55. doi:
10.1021/ci400716y. Epub 2014 Feb 21. PMID: 24521170.

13: Herrera-Ibatá DM, Pazos A, Orbegozo-Medina RA, Romero-Durán FJ, González-
Díaz H. Mapping chemical structure-activity information of HAART-drug cocktails
over complex networks of AIDS epidemiology and socioeconomic data of U.S.
counties. Biosystems. 2015 Jun;132-133:20-34. doi:
10.1016/j.biosystems.2015.04.007. Epub 2015 Apr 24. PMID: 25916548.

14: Quevedo-Tumailli VF, Ortega-Tenezaca B, González-Díaz H. Chromosome Gene
Orientation Inversion Networks (GOINs) of Plasmodium Proteome. J Proteome Res.
2018 Mar 2;17(3):1258-1268. doi: 10.1021/acs.jproteome.7b00861. Epub 2018 Feb 5.
PMID: 29336158.

15: Barreiro E, Munteanu CR, Cruz-Monteagudo M, Pazos A, González-Díaz H. Net-
Net Auto Machine Learning (AutoML) Prediction of Complex Ecosystems. Sci Rep.
2018 Aug 17;8(1):12340. doi: 10.1038/s41598-018-30637-w. PMID: 30120369; PMCID:
PMC6098100.

16: Nocedo-Mena D, Cornelio C, Camacho-Corona MDR, Garza-González E, Waksman de
Torres N, Arrasate S, Sotomayor N, Lete E, González-Díaz H. Modeling
Antibacterial Activity with Machine Learning and Fusion of Chemical Structure
Information with Microorganism Metabolic Networks. J Chem Inf Model. 2019 Mar
25;59(3):1109-1120. doi: 10.1021/acs.jcim.9b00034. Epub 2019 Mar 4. PMID:
30802402.

17: Diéguez-Santana K, Casañola-Martin GM, Green JR, Rasulev B, González-Díaz H.
Predicting Metabolic Reaction Networks with Perturbation-Theory Machine Learning
(PTML) Models. Curr Top Med Chem. 2021;21(9):819-827. doi:
10.2174/1568026621666210331161144. PMID: 33797370.

18: Diéguez-Santana K, Casañola-Martin GM, Torres R, Rasulev B, Green JR,
González-Díaz H. Machine Learning Study of Metabolic Networks <i>vs</i> ChEMBL
Data of Antibacterial Compounds. Mol Pharm. 2022 Jul 4;19(7):2151-2163. doi:
10.1021/acs.molpharmaceut.2c00029. Epub 2022 Jun 7. PMID: 35671399.
